tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/security/PersistedSecuritySettingTest.java

```diff
@@ -43,4 +48,31 @@ public void testNPE() {
       persistedSecuritySetting.getViewRoles();
       persistedSecuritySetting.getEditRoles();
    }
+
+   @Test
+   public void testUpgradeAfterARTEMIS_4582() {
```

Comment:
did you think about writing a compatibility test, perhaps using the functionality?

Reply:
I didn't. The stack-trace on the exception was pretty clear, and since we've dealt with this kind of issue before a quick unit test seemed adequate.